sql/analyzer/resolve_columns.go

```diff
@@ -520,32 +555,85 @@ func resolveColumns(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel
 			return n, transform.SameTree, nil
 		}
 
-		if _, ok := n.(sql.Expressioner); !ok {
-			return n, transform.SameTree, nil
+		var jtNew, left sql.Node
```

Comment:
Would be better to run this new logic as a separate step before the main transform, rather than as part of this one